test/sagemaker_tests/huggingface_tensorflow/training/integration/sagemaker/test_smdataparallel.py

```diff
@@ -0,0 +1,111 @@
+    """
+    Tests SMDataParallel single-node command via script mode
+    """
+    _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+    image_cuda_version = get_cuda_version_from_tag(ecr_image)
+    if Version(image_framework_version) < Version("2.3.1") or image_cuda_version != "cu110":
+        pytest.skip("Data Parallelism is only supported on CUDA 11, and on TensorFlow 2.3.1 or higher")
+
+    # configuration for running training on smdistributed Data Parallel
+    distribution = {'smdistributed': {'dataparallel': {'enabled': True}}}
+
+    instance_type = "ml.p3.16xlarge"
+    instance_count = 1
+
+    estimator = HuggingFace(
+        entry_point='train.py',
+        source_dir=BERT_PATH,
+        role='SageMakerRole',
+        instance_type=instance_type,
+        instance_count=instance_count,
+        image_uri=ecr_image,
+        framework_version=framework_version,
+        py_version=py_version,
+        distribution=distribution,
+        sagemaker_session=sagemaker_session,
+        hyperparameters=hyperparameters,
+        debugger_hook_config=False,  # currently needed
+    )
+
+    estimator.fit(job_name=unique_name_from_base('test-tf-hf-smdp'))
+
+
+@pytest.mark.processor("gpu")
+@pytest.mark.skip_cpu
+@pytest.mark.multinode(2)
+@pytest.mark.integration("hf_smdp_multinode")
+@pytest.mark.model("hf_distilbert")
+@pytest.mark.skip_py2_containers
+@pytest.mark.parametrize('instance_types', ["ml.p3.16xlarge"])
+# Temprarily skipping `ml.p3dn.24xlarge` instance type due to capacity issue in us-west-2
+# TODO: Revert this change asap
+def test_hf_smdp_multi(instance_types, ecr_image, py_version, sagemaker_session, tmpdir, framework_version):
+    """
+    Tests smddprun command via Estimator API distribution parameter
+    """
+    _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+    image_cuda_version = get_cuda_version_from_tag(ecr_image)
+    if Version(image_framework_version) < Version("2.3.1") or image_cuda_version != "cu110":
+        pytest.skip("Data Parallelism is only supported on CUDA 11, and on TensorFlow 2.3.1 or higher")
+
+    distribution = {"smdistributed": {"dataparallel": {"enabled": True}}}
+    instance_type = "ml.p3.16xlarge"
+    instance_count = 2
+
+    estimator = HuggingFace(entry_point='train.py',
+                           source_dir=BERT_PATH,
+                           role='SageMakerRole',
+                           instance_type=instance_type,
+                           instance_count=instance_count,
+                           image_uri=ecr_image,
+                           framework_version=framework_version,
+                           py_version=py_version,
+                           sagemaker_session=sagemaker_session,
+                           hyperparameters=hyperparameters,
+                           distribution=distribution,
+                           debugger_hook_config=False,  # currently needed
```

Comment:
Please create a GitHub Issue here to track the fact that this flag needs to be removed whenever possible, if it isn't possible to delete this right now.

Also, please add a `#TODO: ` statement around this section of code to let any readers know that this needs to be removed. Maybe you could create the Issue first, and link to that in the TODO statement.